### PR TITLE
WQP-1391

### PR DESCRIPTION
**Pull request by dsteinich**

### CODE REVIEW

**Comment by mbucknell** [comment]
Question about types to use. In the NWIS lookups etl I used int and boolean primitive types rather than Integer and Boolean. Should I change that? 

A slightly different question, I also used java.util.Date rather than java.time.LocalDate. This one I think I should change, yes?

**Comment by mbucknell** [comment]
As you may have noticed, I went ahead and hooked up Codacy to this repo. There are ways to configure which rules we would like ignored. We can do it in the tool or better would be figuring out to configuration files that Codacy would then use during it's analysis. We may want a ticket around that too.

**Comment by mbucknell** [comment]
Another question: Should the artifact version be 0.1.0-SNAPSHOT rather than 0.0.1-SNAPSHOT in the pom.xml

File: src/main/java/gov/acwi/wqp/etl/activity/index/BuildActivityIndexes.java [hunk 1]
```diff
@@ -52,11 +52,11 @@
...
-//
-//	@Autowired
-//	@Qualifier("buildActivityOrganizationIndex")
-//	private Tasklet buildActivityOrganizationIndex;
-//
+
+	@Autowired
+	@Qualifier("buildActivityOrganizationIndex")
+	private Tasklet buildActivityOrganizationIndex;
+
```
[the remark this answers is not in the document]
Comment (dsteinich) [reply src/main/java/gov/acwi/wqp/etl/activity/index/BuildActivityIndexes.java]: Yes, JIRA WQP-1400 created

File: src/main/java/gov/acwi/wqp/etl/codes/assemblage/TransformAssemblageTasklet.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	@Autowired
+	public TransformAssemblageTasklet(JdbcTemplate jdbcTemplate,
+			PreparedStatementSetter pss) {
+		this.jdbcTemplate = jdbcTemplate;
+		this.pss = pss;
+	}
+
+	@Override
+	public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
+		jdbcTemplate.query("select * from transform_assemblage(?,?)", pss, new NoopResultSetExtractor());
```
[the remark this answers is not in the document]
Comment (dsteinich) [reply src/main/java/gov/acwi/wqp/etl/codes/assemblage/TransformAssemblageTasklet.java]: At this point, CorePreparedStatementSetter is the only PreparedStatementSetter in the project. Thus it gets autowired in. (The next round of changes may require us to add the @Qualifier annotiation...)

The @StepScope makes the current jobs context available to the class - thus making the data source and schema name available to the CorePreparedStatementSetter (by the annotation in it)

File: src/main/java/gov/acwi/wqp/etl/summaries/activitySum/TransformActivitySum.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+				.tasklet(transformActivitySumTasklet)
+				.build();
+	}
+
+	@Bean
+	public Flow activitySumFlow() {
+		return new FlowBuilder<SimpleFlow>("activitySumFlow")
+				.start(setupActivitySumSwapTableFlow)
+				.next(transformActivitySumStep())
+				.next(buildActivitySumIndexesFlow)
```
[the remark this answers is not in the document]
Comment (dsteinich) [reply src/main/java/gov/acwi/wqp/etl/summaries/activitySum/TransformActivitySum.java]: Step vs. Flow is purely a matter of how much you are attempting to do. The "setup" contains two steps to accomplish. The "transform" is a single step. The "buildIndexes" will also contain multiple steps.

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 4]
```diff
@@ -11,6 +11,10 @@ databaseChangeLog:
       file: "functions/index/changeLog.yml"
       relativeToChangelogFile: true
 
+  - include:
```
[the remark this answers is not in the document]
Comment (dsteinich) [reply src/main/resources/db/changelog/db.changelog-master.yaml]: Just like the create of the batch* tables, these will be run at app startup. During my testing it doesn't appear to add significant time to the startup when there are no changes to be applied.

**Pull request merged**